### ISSUE TITLE
Debian php.ini 

### ISSUE THREAD

**Issue by fischeol** [richardforth/apache2buddy#197]
Hey,
on debian apache uses  /etc/php5/apache2/php.ini instead of  /etc/php5/cli/php.ini
but your call uses the cli php.ini

mentioned it, because memory_limit was changed and your script said it was still -1

/etc/debian_version
8.7

greetings

**Comment by richardforth** [comment]
Ah thanks for that. I'll make the change later.

**Comment by richardforth** [comment]
Hi, 

Just looking at the code, i am not catting any files here, im using the php executable, which may explain why youre seeing this on debian, though im not sure how best to fix this:

```perl

sub detect_php_memory_limit {
	if ( ! $NOINFO) {
		our $apache_proc_php = get_php_setting('/usr/bin/php', 'memory_limit');
		show_info_box(); print "Your PHP Memory Limit (Per-Process) is ${CYAN}".$apache_proc_php." MB${ENDC}.\n";
		if ($apache_proc_php eq "-1") {
			show_advisory_box(); print "You should set a PHP Memory Limit (-1 is ${CYAN}UNLIMITED${ENDC}) which is not recommended.\n";
		}
	}
}

```

If you have any thoughts' I'd be happy to implement - so long as it works cross platform.

**Comment by fischeol** [comment]
Hi Richard,

it seems the path is hardcoded in the apache php module. The only idea at the moment that I have, is looking via dpkg if the packed is installed:

dpkg -L libapache2-mod-php5
/.
/etc
/etc/apache2
/etc/apache2/mods-available
/etc/apache2/mods-available/php5.load
/etc/apache2/mods-available/php5.conf
/etc/php5
/etc/php5/apache2
/etc/php5/apache2/conf.d
[…]

It seems, that the directory is created by the package and copied from the cli to this position afterwards.



Von: Richard Forth [mailto:notifications@github.com]
Gesendet: Mittwoch, 7. Juni 2017 10:16
An: richardforth/apache2buddy <apache2buddy@noreply.github.com>
Cc: Fischer, Oliver <Oliver.Fischer@MKK.de>; Author <author@noreply.github.com>
Betreff: Re: [richardforth/apache2buddy] Debian php.ini (#197)


Hi,

Just looking at the code, i am not catting any files here, im using the php executable, which may explain why youre seeing this on debian, though im not sure how best to fix this:

sub detect_php_memory_limit {

        if ( ! $NOINFO) {

               our $apache_proc_php = get_php_setting('/usr/bin/php', 'memory_limit');

               show_info_box(); print "Your PHP Memory Limit (Per-Process) is ${CYAN}".$apache_proc_php." MB${ENDC}.\n";

               if ($apache_proc_php eq "-1") {

                       show_advisory_box(); print "You should set a PHP Memory Limit (-1 is ${CYAN}UNLIMITED${ENDC}) which is not recommended.\n";

               }

        }

}

If you have any thoughts' I'd be happy to implement - so long as it works cross platform.

**Comment by richardforth** [comment]
Update:

so with php you can specify a config file to use on execution, so I could theoretically see if we are on Debian  / check for existence of a file in /etc/php5/apache/php.ini and then run `php -f /etc/php5/apache/php.ini` or something  - when searching for php environment variables, though its going to take a bit of recoding to get that to work, and I don't have a lot of free time for coding anymore, but I have put this here so I dont forget about it.

**Comment by richardforth** [comment]
or

our $apache_proc_php = get_php_setting('/usr/bin/php', 'memory_limit');


Make this subroutine accespt three variables instead of two:

1. the binary file
2. *NEW* path to config  - or default to /etc/php.ini
3. what we are looking for.

However the new 'default' of /etc/php.ini would also break inder debian because it has to be different! Arrrgh! 

will need to think about this

**Comment by richardforth** [comment]
root@debian9:~# grep -HiR ^mem /etc/php/7.0/
/etc/php/7.0/cli/php.ini:memory_limit = -1
/etc/php/7.0/apache2/php.ini:memory_limit = 256M <-- I changed this to 256, default was 128

**Comment by richardforth** [comment]
[ -- ] Your PHP Memory Limit (Per-Process) is -1 MB.
[ @@ ] You should set a PHP Memory Limit (-1 is UNLIMITED) which is not recommended.

picking up wrong config indeed, this is on debian 9  so issue still persists,

**Comment by richardforth** [comment]
```
root@debian9:~/apache2buddy# php -r 'phpinfo(1);' | grep -i config
Configuration File (php.ini) Path => /etc/php/7.0/cli
Loaded Configuration File => /etc/php/7.0/cli/php.ini
```

**Comment by richardforth** [comment]
added some verboseness, 

VERBOSE: Loaded Configuration File => /etc/php/7.0/cli/php.ini
[ -- ] Your PHP Memory Limit (Per-Process) is -1 MB.
[ @@ ] You should set a PHP Memory Limit (-1 is UNLIMITED) which is not recommended.

**Comment by richardforth** [comment]
VERBOSE: PHP: Loaded Configuration File => /etc/php/7.0/cli/php.ini
[ -- ] Your PHP Memory Limit (Per-Process) is -1 MB.
[ @@ ] You should set a PHP Memory Limit (-1 is UNLIMITED) which is not recommended.

**Comment by richardforth** [comment]
VERBOSE: PHP: Loaded Configuration File => /etc/php/7.0/cli/php.ini
VERBOSE: PHP: Attempting to find real apache php.ini file...
[ -- ] Your PHP Memory Limit (Per-Process) is -1 MB.
[ @@ ] You should set a PHP Memory Limit (-1 is UNLIMITED) which is not recommended.

**Comment by richardforth** [comment]
Almost....

VERBOSE: PHP: Loaded Configuration File => /etc/php/7.0/cli/php.ini
VERBOSE: PHP: Attempting to find real apache php.ini file...
VERBOSE: PHP: Real apache php.ini file is /etc/php/7.0/apache2/php.ini, using that...
[ -- ] Your PHP Memory Limit (Per-Process) is -1 MB.
[ @@ ] You should set a PHP Memory Limit (-1 is UNLIMITED) which is not recommended.

**Comment by richardforth** [comment]
# php -c /etc/php/7.0/apache2/php.ini -r 'phpinfo(1);' | grep -i config | grep -i loaded
Loaded Configuration File => /etc/php/7.0/apache2/php.ini


Ok....fingers crossed

**Comment by richardforth** [comment]
```
VERBOSE: PHP: Loaded Configuration File => /etc/php/7.0/cli/php.ini
VERBOSE: PHP: Attempting to find real apache php.ini file...
VERBOSE: PHP: Real apache php.ini file is /etc/php/7.0/apache2/php.ini, using that...
[ -- ] Your PHP Memory Limit (Per-Process) is 256 MB.
```

**Comment by richardforth** [comment]
will need to check if this works across other debians and still works on redhat...

**Comment by richardforth** [comment]
fixed in the following commit(s):

  | richardforth | Updated apache2buddy.pl |   |   | c7a7e2b
  |   | richardforth | Closes #205 |   |   | 93e0656
  |   | richardforth | Closes #205 |   |   | 0921e8b
  |   | richardforth | Closes #202 |   |   | 3f1f049
  |   | richardforth | Closes #206, Closes #207

**Comment by richardforth** [comment]
also works in debian 8 too:

```
VERBOSE: PHP: Loaded Configuration File => /etc/php5/cli/php.ini
VERBOSE: PHP: Attempting to find real apache php.ini file...
VERBOSE: PHP: Real apache php.ini file is /etc/php5/apache2/php.ini, using that...
[ -- ] Your PHP Memory Limit (Per-Process) is 128 MB.
```

Note that Debian 7 support was dropped, I may look into supporting it again but at the moment there are unresolved issues with its perl modules out-of-the-box
